_data-prepper/pipelines/configuration/processors/anomaly-detector.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: default
+title: Anomaly detector
+parent: Processors
+grand_parent: Pipelines
+nav_order: 45
+---
+
+# Anomaly detector
+
+## Overview
+
+The `Anomaly detector` processor takes structured data and runs anomaly detection algorithms on fields you can configure in the data. The data must be either an integer or real number in order for the the anomaly detection algorithm to detect anomalies. We recommend that you deploy the `Aggregate` processor in a pipeline before the `Anomaly detector` processor to achieve the best results.  This is because the `Aggregate` processor aggregates events with same keys onto the same host.
+
+This processor uses the `random_cut_forest` mode to detect anomalies. 
+
+
+## Configuration
+
+Configuration for this processor involves specifying a key and specifying options for the mode. You can use the following options to configure the `Anomaly detector` processor.
+
+| Name | Required | Description |
+| :--- | :--- | :--- |
+| `keys` | Yes | A non-ordered `List<String>` which are used as inputs to the ML algorithm to detect anomalies in the values of the keys in the list. At least one key is required.
+| `mode` | Yes |  The machine learning (ML) algorithm (or model) to use to detect anomalies. One of the existing [Modes](#modes) must be provided. See [random_cut_forest](#random_cut_forest).
+
+### Keys
+
+
+### random_cut_forest mode
+
+<!--- Add description for the random_forest_cut mode.--->
+
+| Name | Description |
+| :--- | :--- |
+| `random_cut_forest` | Processes events using Random Cut Forest ML algorithm to detect anomalies. After passing a bunch of events with `latency` value between 0.2 and 0.3 are passed through the anomaly detector, when an event with `latency` value 11.5 is sent, the following anomaly event will be generated. See [Random Cut Forest (RCF) Algorithm](https://docs.aws.amazon.com/sagemaker/latest/dg/randomcutforest.html) for more details.| 
+
+
+
+ ```json
```

Comment:
@dlvenable @asifsmohammed What is the purpose of this JSON file? How are we supposed to use it? And how does it relate to the text that follows on the next line?

Reply:
The purpose of the JSON snippet is to show what happens in the anomaly detector processor when that input is received by the processor. We use the similar style of example snippet in other processors too. The text following explains what happens to the event (mentioned in json snippet) when it passes through the processor.